php_phongo.c

```diff
@@ -850,7 +850,13 @@ int phongo_execute_command(mongoc_client_t* client, php_phongo_command_type_t ty
 			return false;
 	}
 	if (!result) {
-		phongo_throw_exception_from_bson_error_t(&error TSRMLS_CC);
+		if (error.domain == MONGOC_ERROR_SERVER || error.domain == MONGOC_ERROR_WRITE_CONCERN) {
+			zend_throw_exception(php_phongo_commandexception_ce, error.message, error.code TSRMLS_CC);
+			php_phongo_bson_to_zval(bson_get_data(&reply), reply.len, return_value);
```

Comment:
The third argument to `php_phongo_bson_to_zval()` is a `zval *` or `zval **` for PHP 7 and 5, respectively. See: https://github.com/mongodb/mongo-php-driver/blob/5bcfdcec81a3802f8f7c8e4c06ecdceb39ceb70f/src/bson.c#L986

```c
#if PHP_VERSION_ID >= 70000
	php_phongo_bson_to_zval(bson_get_data(&reply), reply.len, return_value);
#else
	php_phongo_bson_to_zval(bson_get_data(&reply), reply.len, &return_value);
#endif
```

That may do the trick.

Reply:
Ah, this actually leaks because `return_value` is already allocated, and `php_phongo_bson_to_zval_ex()` tries to allocate it again for PHP 5.x: https://github.com/mongodb/mongo-php-driver/blob/5bcfdcec81a3802f8f7c8e4c06ecdceb39ceb70f/src/bson.c#L1028

Solution would be to go back to the Command class' debug info handler. Declare either a `zval zv` or `zval *zv` for PHP 7 and 5, respectively, and then we can just call `**php_phongo_bson_to_zval(bson_get_data(&reply), reply.len, &zv)` for all versions.